go/nbs/dynamo_manifest.go

```diff
@@ -76,25 +79,28 @@ func validateManifest(item map[string]*dynamodb.AttributeValue) (valid, hasSpecs
 	if item[nbsVersAttr] != nil && item[nbsVersAttr].S != nil &&
 		StorageVersion == *item[nbsVersAttr].S &&
 		item[versAttr] != nil && item[versAttr].S != nil &&
+		item[lockAttr] != nil && item[lockAttr].B != nil &&
 		item[rootAttr] != nil && item[rootAttr].B != nil {
-		if len(item) == 5 && item[tableSpecsAttr] != nil && item[tableSpecsAttr].S != nil {
+		if len(item) == 6 && item[tableSpecsAttr] != nil && item[tableSpecsAttr].S != nil {
 			return true, true
 		}
-		return len(item) == 4, false
+		return len(item) == 5, false
 	}
 	return false, false
 }
 
-func (dm dynamoManifest) Update(specs []tableSpec, root, newRoot hash.Hash, writeHook func()) (actual hash.Hash, tableSpecs []tableSpec) {
+func (dm *dynamoManifest) Update(specs []tableSpec, root, newRoot hash.Hash, writeHook func()) (actual hash.Hash, tableSpecs []tableSpec, err error) {
 	tableSpecs = specs
 
+	lock := generateLockHash(newRoot, specs)
```

Comment:
Hm. generally, `newFoo` is, in our code, a function that constructs a thing of type `Foo`. Since `Lock` isn't a type, here, that naming convention seems wrong to me.